### PR TITLE
Consolidate Bitmap(Type, string) constructor

### PR DESCRIPTION
**Pull request by hughbe**
Contributes to #22221

Also, fix a NRE if `type == null` - this is a breaking change, but one I think is acceptable

### CODE REVIEW

File: src/libraries/System.Drawing.Common/tests/mono/System.Drawing/BitmapTests.cs [hunk 1]
```diff
@@ -1219,78 +1219,6 @@ public void XmlSerialization()
...
-        public void BitmapTypeStringCtor1()
-        {
-            Assert.Throws<NullReferenceException>(() => new Bitmap((Type)null, "mono"));
-        }
-
-        [ConditionalFact(Helpers.IsDrawingSupported)]
-        public void BitmapTypeStringCtor2()
-        {
-            AssertExtensions.Throws<ArgumentNullException, ArgumentException>("resource", null, () => new Bitmap(typeof(Bitmap), null));
-        }
```
Comment (stephentoub) [inline src/libraries/System.Drawing.Common/tests/mono/System.Drawing/BitmapTests.cs:1292]: I understand why the NRE test is being deleted, but why the others?  Are these duplicates, and if so, of what?

File: src/libraries/System.Drawing.Common/src/System/Drawing/Bitmap.cs [hunk 2]
```diff
@@ -49,7 +49,33 @@ public Bitmap(string filename, bool useIcm)
...
+            }
+
+            Stream? stream = type.Module.Assembly.GetManifestResourceStream(type, resource);
+            if (stream == null)
+            {
+                throw new ArgumentException(SR.Format(SR.ResourceNotFound, type, resource));
+            }
+
+            return stream;
+        }
```
Comment (stephentoub) [inline src/libraries/System.Drawing.Common/src/System/Drawing/Bitmap.cs:78]: Refactoring seems reasonable. Thanks.

File: src/libraries/System.Drawing.Common/tests/mono/System.Drawing/BitmapTests.cs (hunk 1 above)
Comment (hughbe) [reply src/libraries/System.Drawing.Common/tests/mono/System.Drawing/BitmapTests.cs:1292]: - `BitmapImageCtor`: dupe of `Ctor_NullImageWithoutSize_ThrowsNullReferenceException`
- `BitmapImageSizeCtor` and `BitmapImageIntIntCtor`: dupe of `Ctor_NullImageWithSize_ThrowsArgumentNullException`
- `BitmapIntIntGraphicCtor`: dupe of `Ctor_NullGraphics_ThrowsArgumentNullException`
- `BitmapIntIntPixelFormatCtor`: dupe of `Ctor_InvalidWidth_ThrowsArgumentException` and `Ctor_InvalidHeight_ThrowsArgumentException`
- `BitmapStreamCtor` and `BitmapStreamBoolCtor`: dupe of `Ctor_NullStream_ThrowsArgumentNullException`
- `BitmapStringCtor` and `BitmapTypeStringCtor2`: dupe of `Ctor_NullFilePath_ThrowsArgumentNullException`

**Comment by hughbe** [comment]
I think the test failures are unrelated

**Review by safern — APPROVED** [review]
Seems reasonable. Thanks.

**Comment by hughbe** [comment]
@safern could I ask you to see about getting some of the libgdiplus PRs merged in? I have some spare time on my hands since we're legally not allowed to leave the house here in the UK.. !

**Comment by safern** [comment]
Sorry, @hughbe I was actually on vacation. Now I'm back.

**Pull request merged**